corehq/form_processor/models/forms.py

```diff
@@ -839,15 +839,17 @@ def commcare_version(self):
 
 
 class TempFormCache:
-    def __init__(self, domain):
-        self.domain = domain
+    def __init__(self):
         self.cache = {}
 
     def get_forms(self, form_ids):
-        cached_forms = [self.cache[form_id] for form_id in form_ids if form_id in self.cache]
-        get_forms = XFormInstance.objects.get_forms([form_id for form_id in form_ids if form_id not in self.cache],
-                                                    self.domain, ordered=True)
-        forms = cached_forms + get_forms
+        forms = [self.cache[form_id] for form_id in form_ids if form_id in self.cache]
+        not_cached_forms = [form_id for form_id in form_ids if form_id not in self.cache]
+        if not_cached_forms:
+            retrieved_forms = XFormInstance.objects.get_forms(not_cached_forms, ordered=True)
+            for form in retrieved_forms:
+                self.cache[form.form_id] = form
+            forms += retrieved_forms
```

Comment:
It doesn't look like this branch is covered by tests. What do you think of adding a test that covers it?